src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -625,22 +625,21 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
 
       def getInactiveBakersWithTaggedAccounts(
           taggedAccounts: List[BlockTagged[Map[AccountId, Account]]]
-      ): Future[List[(BlockTagged[Map[AccountId, Account]], List[Tables.AccountsHistoryRow])]] =
+      ): Future[List[(BlockTagged[Map[AccountId, Account]], List[Tables.AccountsRow])]] =
         Future.traverse(taggedAccounts) { blockTaggedAccounts =>
           if (blockTaggedAccounts.blockLevel % lorreConf.blockRightsFetching.cycleSize == 1) {
             tezosNodeOperator.fetchActiveDelegates(taggedAccounts.map(x => (x.blockLevel, x.blockHash))).flatMap {
               activeBakers =>
                 db.run {
                   TezosDb
-                    .getInactiveBakersFromAccountsHistoryByBlock(
-                      blockTaggedAccounts.blockLevel,
+                    .getInactiveBakersFromAccounts(
                       activeBakers.toMap.apply(blockTaggedAccounts.blockHash)
```

Comment:
optimize: this should've been catched in earlier reviews but nonetheless...
We want to convert the list to a Map only once for each `flatMap` call... instead we're doing it on every call to the TezosDb method, times the number of flatMap calls.

It might be premature optimization, yet the cost to rewrite is minimal, while each `toMap` will traverse the whole list again to build the new map object, which is also adding to the global GC pressure.

Reply:
sure, changed